syntax = "proto3";
package monitor;

import "google/protobuf/timestamp.proto";
import "google/protobuf/empty.proto";

message ServerStatus {
    int32 host_count = 1;
    repeated ProjectStatus projects = 2;
    google.protobuf.Timestamp time = 3;
}

message ProjectStatus {
    string id = 1;
    string name = 2;
    string description = 3;
    int32 healthy_count = 4;
    int32 total_count = 5;
}

service MonitorPublicService {
    rpc GetStatus (google.protobuf.Empty) returns (ServerStatus);
}
